### PR TITLE
[produce] `app_version` option is ignored.

### PR DESCRIPTION
**Pull request by xai3**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Description
`Version` removed from the form of creating new application from some day.
So I removed `version` argument from `Spaceship::Tunes::Application.create`.
And I added code to chage the version after the creating new application.

| before | after |
|---|---|
| new_ios_app_filled |  |

### Motivation and Context
Issue: #8469

### CODE REVIEW

File: spaceship/lib/spaceship/tunes/tunes_client.rb
Comment (KrauseFx) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: This would be a breaking change when someone calls _spaceship_ scripts directly. Is there a good way in Ruby to deprecate a parameter?

Comment (yukiasai) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: You're right. I'll fix it!

**Comment by KrauseFx** [comment]
Thanks for updating the PR, this looks much better now 👍

File: produce/lib/produce/itunes_connect.rb
Comment (KrauseFx) [inline produce/lib/produce/itunes_connect.rb]: Spelling mistake, this should be `Ensuring` 👍

File: produce/lib/produce/itunes_connect.rb
Comment (KrauseFx) [inline produce/lib/produce/itunes_connect.rb]: ❤️

**Review by KrauseFx — APPROVED** [review]
Amazing, this looks great to me, I'd love to get another +1 from someone else before merging it though, ideally someone can test it on the real iTunes Connect? 👍

File: produce/lib/produce/itunes_connect.rb
Comment (yukiasai) [inline produce/lib/produce/itunes_connect.rb]: Oops..

File: spaceship/lib/spaceship/tunes/application.rb
Comment (getaaron) [inline spaceship/lib/spaceship/tunes/application.rb]: A feature marked as "deprecated" should still work. But here we're just throwing out `version` entirely.

I think we should raise an exception if `version` is passed since we're ignoring it.

Comment (KrauseFx) [inline spaceship/lib/spaceship/tunes/application.rb]: But until now we also just ignored the value, as the API ignored the value, right?

**Comment by yukiasai** [comment]
I signed it!

**Review by KrauseFx — APPROVED** [review]

**Pull request merged**
